app/src/main/java/mozilla/lockbox/extensions/ViewNode+.kt

```diff
@@ -15,21 +17,21 @@ private fun AssistStructure.ViewNode.dumpNode(sb: StringBuilder = StringBuilder(
     val name = htmlInfo?.tag ?: className?.split('.')?.last() ?: "unknown"
 
     var attrs = listOf(
-        Pair("idEntry", idEntry ?: ""),
-        Pair("idPackage", idPackage ?: ""),
-        Pair("idType", idType ?: ""),
-        Pair("webDomain", webDomain ?: ""),
-        Pair("hint", hint ?: ""),
-        Pair("autofillValue", autofillValue?.textValue ?: ""),
-        Pair("autofillHints", autofillHints?.joinToString(", ") ?: ""),
-        Pair("autofillOptions", autofillOptions?.joinToString(", ") ?: "")
+        Pair("idEntry", idEntry ?: emptyString),
+        Pair("idPackage", idPackage ?: emptyString),
+        Pair("idType", idType ?: emptyString),
+        Pair("webDomain", webDomain ?: emptyString),
+        Pair("hint", hint ?: emptyString),
+        Pair("autofillValue", autofillValue?.isText ?: autofillValue?.textValue ?: emptyString),
+        Pair("autofillHints", autofillHints?.joinToString(", ") ?: emptyString),
+        Pair("autofillOptions", autofillOptions?.joinToString(", ") ?: emptyString)
     )
 
     htmlInfo?.attributes?.let { attributes ->
         attrs += attributes.map { Pair(it.first, it.second) }
     }
 
-    attrs = attrs.filter { it.second != "" && it.second != "null" }
+    attrs = attrs.filter { it.second != emptyString && it.second != "null" }
```

Comment:
Given that we're filtering here for the placeholders `emptyString` can we get away with having `String?` here instead of `String`?

Also, if you're touching this code, please could you consider replacing `Pair(x, y)` with `x to y`.